src/mcp/server/fastmcp/utilities/func_metadata.py

```diff
@@ -232,9 +275,24 @@ def func_metadata(
                 WithJsonSchema({"title": param.name, "type": "string"}),
             ]
 
+        # Get the typed annotation first to check if it's optional
+        typed_annotation = _get_typed_annotation(annotation, globalns)
+        
+        # Determine the default value for the field
+        if param.default is not inspect.Parameter.empty:
+            # Parameter has an explicit default value
+            default_value = param.default
+        elif _is_optional_type(typed_annotation):
+            # Parameter is Optional[T] or Union[T, None] without explicit default
+            # Set default to None to make it truly optional in the schema
+            default_value = None
+        else:
+            # Required parameter
+            default_value = PydanticUndefined
```

Comment:
I might be wrong, but this feels like we might be using Pydantic incorrectly if we need to do this kind of type introspection.

Any thoughts on this @Kludex?

Reply:
I don't think adding a default value is needed here...